### PR TITLE
CI: Update Clickhouse server image tag to 18.14

### LINKED ISSUE #2172: CI: [Clickhouse] clickhouse-server:18.12 doesn't have toDayOfYear support
The PR https://github.com/ClickHouse/ClickHouse/pull/3146 adds toDayOfYear support was merged at Sep 17th, so the docker image tag more close to this date is v18.14 

would be good this tag to allow support for toDayOfYear operation.

### PR DESCRIPTION
**Pull request by xmnlab**
Fix #2172

### CODE REVIEW

**Comment by xmnlab** [comment]
this is ready for review! thanks!

**Comment by jreback** [comment]
ok, can you get this passing on current master (I think something just broke). Also does this change the support of clickhouse?

**Comment by xmnlab** [comment]
> ok, can you get this passing on current master (I think something just broke). 

@jreback I will rebase on top of master to check that

> Also does this change the support of clickhouse?

sorry I didn't understand, I just changed that because I am working in a new PR to add support to DayOfYear operation ... could you give me some examples ?

**Comment by jreback** [comment]
> > ok, can you get this passing on current master (I think something just broke).
> 
> @jreback I will rebase on top of master to check that
> 
> > Also does this change the support of clickhouse?
> 
> sorry I didn't understand, I just changed that because I am working in a new PR to add support to DayOfYear operation ... could you give me some examples ?

I think something is broken in the CI currently. Please get this PR passing and can merge.

**Comment by xmnlab** [comment]
sounds good .. I am working on that :)

**Comment by xmnlab** [comment]
@jreback CI is green now.

It seems each `make docker_run` and `docker_docs_run` was building the image each time they were called. 

also I split conda update and conda install ... into 2 `RUN` it uses less space on disk

File: Makefile [hunk 1]
```diff
@@ -176,8 +176,8 @@ doc: builddoc docclean
 
 # Targets for run commands inside ibis and ibis-docs containers
 
-docker_run: build
+docker_run:
```
Comment (jreback) [inline Makefile:179]: if these are not using the current build what are they using?

**Comment by jreback** [comment]
please make a new PR with the CI changes only -

**Comment by xmnlab** [comment]
done: #2176

**Comment by jreback** [comment]
k th anks

**Pull request merged**